hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -122,15 +148,18 @@ sealed abstract class Block extends Product with AutoLocated:
   // Note that this returns the definitions in reverse order, with the bottommost definiton appearing
   // last. This is so that using defns.foldLeft later to add the definitions to the front of a block, 
   // we don't need to reverse the list again to preserve the order of the definitions.
-  def floatOutDefns =
+  def floatOutDefns(
+    ignore: Defn => Bool = _ => false, 
+    preserve: Defn => Bool = _ => false) =
```

Comment:
Better to distinguish parameters from body using indentation:
```suggestion
      ignore: Defn => Bool = _ => false,
      preserve: Defn => Bool = _ => false,
  ) =
```